---
title: "Supplemental Methods"
author: "from Angelini, Steele and O'Brien. Gene expression during development of a dispersal polyphenism identifies regulators and effectors of growth in muscle and germline. *Integrative & Comparative Biology*."
date: "Updated `r format(Sys.time(), '%d %B %Y')`"
output:
  html_document:
    toc: true
    toc_depth: 2
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

> **This is a pre-submission DRAFT**.  Some of this text will be moved into the Methods and Results section of the manuscript's main text. Other text may remain in supplemental materials. 
>
> Details that may be important now, but should likely be removed before publication are given as block quotes.
>
> Code is available at https://github.com/aphanotus/morphDE. While any process is running on nscc, you can monitor it's progress indirectly by checking the CPU usage on [NSCC node 28](http://schupflab.colby.edu/mrtg/nscc/nscc-n28.load.html).

In order to characterize the development of polyphenic traits in the wings, flight muscles and gonads of the soapberry bug, *Jadera haematoloma*, we sequenced transcriptomes from a multidimensional matrix of life stage, tissue, sex, food regime and host population.

<a name="sampling-design"><a/>

### Sampling design

Transcriptome studies proceeded through three phases. Initially, we sampled whole bodies from 12 individuals, including three biological replicates of each sex and morph combination from the Plantation Key, FL population (phase 1). Next we sampled dorsal thorax and gonads from three individuals of each sex and morph combination from the Plantation Key, FL and Aurora, CO populations (phase 2). Samples from these phases were prepared as full-length 150-bp paired-end Illumina libraries and used to assembly a reference transcriptome. 

The final phase utilized 3'-tag sequencing with the exclusive goal of quantifying gene expression. The sampling design at this phase included two stages (fifth instars and nascent adults), two tissues (dorsal thorax and gonads), two sexes, long- and short-wing adult morphs, high and low food regimes, four populations (collected from Key Largo, FL, Plantation Key, FL, Aurora, CO, and Frederick, MD), and three biological replicates of each unique combination of the preceding factors. Thus the full matrix contained 288 samples. Seven samples were lost or did not pass library quality controls, resulting in 281 samples in the analysis.

<a name="shape-analysis"><a/>

### Shape analysis

Most *J. haematoloma* can be qualitatively identified as "long winged" or "short winged" based on the shape of the wings and whether their distal tips extend beyond the posterior of the abdomen when folded up at rest. However, closer inspection reveals that some individuals have intermediate shapes and lengths. We reasoned that gene expression was likely to also follow a more continuous distribution. Therefore, rather than relying on a categorical assignment of specimens into morphs, gene expression was modeled based on the quantification of wing shapes. (Categorical modeling by morph was also performed, and the two approaches identified similar lists of top DEGs.) Thoracic muscles are required for flight, therefore we also tested for correlations between gene expression and thorax shape.  

Bugs were anesthetized using CO~2~ and imaged on a trinocular stereo microscope (VWR International, Radnor, Pennsylvania, USA) with a Moticam 5 digital camera. Dorsal and ventral images of each specimen were recorded with a millimeter-scale ruler to scale pixel measurements. ImageJ was used to place landmarks.

Adult wing shape was quantified as described by Fawcett et al. (2018). 

**Table SX**: Wing landmark positions 

| landmark | position                                                     |
|:-------- |:------------------------------------------------------------ |
| 1        | Bifurcation of the radial and medial veins                   |
| 2*       | Maximum curvature of clavus, near the posterior tip of the mesonotum |
| 3-4*     | Two equally spaced semilandmarks along the claval margin, anterior of landmark 2 |
| 5        | Bifurcation of the radial vein and R2 crossvein              |
| 6        | Intersection of the R2 crossvein and the medial vein         |
| 7        | Intersection of the cubitus vein and the first anal vein     |
| 8*       | One semilandmark along the claval margin, between landmarks 2 and 7 |
| 9        | Intersection of the medial vein and the posterior crossvein  |
| 10       | Intersection of the radial vein and the anterior crossvein   |
| 11       | Terminus of the radial-medial crossvein                      |
| 12       | First branch of the anterior membrane vein                   |
| 13*      | Maximum posterior-distal curvature of the membrane region of the wing (roughly lateral to landmark 11 when held in place on the bug) |
| 14-16*   | Three equally spaced semilandmarks along the posterior wing margin, between landmarks 7 and 13 |
| 17-21*   | Five equally spaced semilandmarks along the distal wing margin, between landmarks 11 and 13 |
| 22-24*   | Three equally spaced semilandmarks along the anterior wing margin, between landmarks 1 and 11 |
|          | * semilandmark                                               |

This system of landmarks was adapted to quantify juvenile wing pad shapes as described in Table SX.

**Table SX**: Juvenile wing pad landmark positions.

| landmark | position                                                     |
|:-------- |:------------------------------------------------------------ |
| 1        | Bifurcation of the radial and medial veins                   |
| 2        | Medial base of the wing pad                                  |
| 3-4*     | Two equally spaced semilandmarks along the base of the wing pad, at its attachment to the body, between landmarks 1 and 2 |
| 5        | Bifurcation of the radial vein and R2 crossvein              |
| 6        | Intersection of the R2 crossvein and the medial vein         |
| 7        | Intersection of the cubitus vein and the first anal vein     |
| 8*       | One semilandmark along the claval margin, between landmarks 2 and 7 |
| 9        | Intersection of the medial vein and the posterior crossvein  |
| 10       | Intersection of the radial vein and the anterior crossvein   |
| 11       | Terminus of the radial-medial crossvein                      |
| 12*      | Maximum posterior-distal curvature of the wing (roughly lateral to landmark 11) |
| 13*      | One equally spaced semilandmark along the posterior wing margin, between landmarks 7 and 12 |
| 14-18*   | Five equally spaced semilandmarks along the distal wing margin, between landmarks 12 and 11 |
| 19-21*   | Three equally spaced semilandmarks along the anterior wing margin, between landmarks 1 and 11 |
|          | * semilandmark                                               |

Thorax shapes of juvenile and adult bugs were digitized using the same set of landmarks (Table SX) .

**Table SX**: Thorax landmark positions.

| landmark | position                                             |
|:-------- |:---------------------------------------------------- |
| 1        | Anterior right of the pronotum, as viewed from above |
| 2        | Posterior right of the pronotum                      |
| 3        | Posterior left of the pronotum                       |
| 4        | Anterior left of the pronotum                        |
| 5        | Anterior midline of the pronotum                     |
| 6        | Anterior left of the mesonotum                       |
| 7        | Anterior midline of the mesonotum                    |
| 8        | Anterior right of the mesonotum                      |
| 9        | Posterior of the mesonotum                           |

|
|

Variation in shapes was analyzed using the geometric morphometric methods implemented in the R packages geomorph and borealis. Generalized Procrustes analysis with partial Procrustes superimposition was preformed using minimized bending energy. The shape of individual specimens was quantified along the first principle component axis (Fig. SX,SX). Procrustes ANOVA with permutation was used to assess hypotheses for patterns of shape variation among the aligned specimens, using 10,000 iterations.

![](https://i.imgur.com/QVa7Q9i.jpg)

**Figure SX**. Morphospace plotting the first two principle component axes for the shapes of juvenile wing pads (A) and adult wings (B). Specimens are color-coded by sex and morph (for adults), and colored convex hulls show the space occupied by these groups. Deformation grids illustrate shapes at the extremes of each axis. (Anterior is up; distal is right.) The percent of total shape variance described by each principle component in given on the axis. 

![](https://i.imgur.com/1SD7sZe.jpg)

**Figure SX**. Morphospace plotting the first two principle component axes for the shapes of juvenile (A) and adult pronotum and mesonotum (B). Specimens are color-coded by sex and morph (for adults), and colored convex hulls show the space occupied by these groups. Deformation grids illustrate shapes at the extremes of each axis. (Anterior is up.) The percent of total shape variance described by each principle component in given on the axis. 




<a name="nucleic-acid-extractions"><a/>

### Nucleic acid extractions

Individuals were collected from laboratory cultures, anesthetized using CO<sub>2</sub> exposure, photographed, and flash frozen in liquid nitrogen. Samples were stored at -80˚C until further processing. The thorax was removed from head and abdomen with a clean scalpel blade, and the dorsal thorax was separated above the legs while tissue was still frozen. Gonads were then dissected from the abdomen of adults. For juveniles, the entire abdomen was included. Nucleic acid extractions used the Invitrogen PureLink RNA extraction kit.

<a name="sequencing-and-assembly-of-a-reference-transcriptome"><a/>

### Sequencing and assembly of a reference transcriptome

Samples used for transcriptome assembly (phases 1 and 2) were delivered on dry ice to Beckman Coulter Genomics (Danvers, Massachusetts) for poly-A selection, preparation of 125 bp paired-end TruSeq libraries, and sequencing using Illumina HiSeq. Libraries from phases 1 and 2 were sequenced separately. Reads from all samples were trimmed using [Trimmomatic](http://www.usadellab.org/cms/?page=trimmomatic) version 0.33 ([Bolger et al. 2014](https://doi.org/10.1093/bioinformatics/btu170)) with the parameters `ILLUMINACLIP:/export/local/src/Trimmomatic-0.33/adapters/TruSeq3-SE.fa:2:30:10 HEADCROP:12 LEADING:3 TRAILING:3 SLIDINGWINDOW:3:15 MINLEN:50`. The resulting reads were assembled into transcripts by [Trinity](http://trinityrnaseq.github.io/) version 2.0.6 ([Grabherr et al. 2011](https://www.nature.com/articles/nbt.1883)). Before annotation, potentially redundant transcripts with greater than 90% identity were collapsed using CD-HIT-EST version 4.6 ([Li & Godzik 2006](https://doi.org/10.1093/bioinformatics/btl158)).  

```bash
cd /research/drangeli/phase2_BCG_RNAseq/Jhae_GU_trinity_assembly/
cd-hit-est -i GU.Trinity.fa -o GU.Trinity.c90.fa -c 0.90 -n 10 -d 0 -M 64000 -T 12
```
> This took about 3.5 days on node 26.

[TransDecoder](https://github.com/TransDecoder/TransDecoder/wiki) identified coding sequences. At each stage, [BUSCO](https://busco.ezlab.org/) version 5.2.2 ([Manni et al. 2021](https://doi.org/10.1093/molbev/msab199)) was used to assess transcriptome completeness.

```bash
busco --in GU.Trinity.fa \
--lineage_dataset /research/drangeli/DB/busco.lineages/hemiptera_odb10 \
--out BUSCO.report.for.GU \
--augustus --augustus_species fly --mode transcriptome --cpu 24
```

Consolidation by `cd-hist-est` dramatically reduced the number of sequences flagged as duplicates (Table S1).

**Table S1**: Details of transcriptome datasets at each stage of processing.

| dataset               | contigs | BUSCO short summary                           |
|:--------------------- | -------:|:--------------------------------------------- |
| initial assembly      | 549,485 | C:97.0%[S:54.2%,D:42.8%],F:1.3%,M:1.7%,n:2510 |
| initial assembly CDS  | 85,937  |                                               |
| 90% consolidation     | 395,125 | C:96.6%[S:77.7%,D:18.9%],F:1.4%,M:2.0%,n:2510 |
| 90% consolidation CDS | 50,771  | C:94.4%[S:77.8%,D:16.6%],F:2.2%,M:3.4%,n:2510 |

<a name="annotation"><a/>

### Annotation of the transcriptome

Transcript annotation was performed by EnTAP ([Hart et al. 2019](https://doi.org/10.1111/1755-0998.13106)) using the NCBI invertebrate RefSeq dataset, SwissProt and Trembl databases. EnTAP assigns sequence names and GO terms based on similarity.

```bash
EnTAP --runP --threads 24 \
-i /research/drangeli/phase2_BCG_RNAseq/Jhae_GU_trinity_assembly/GU.Trinity.c90.fa \
-d /export/groups/drangeli/entap_config/bin/invert.dmnd \
-d /export/groups/drangeli/entap_config/bin/uniprot_sprot.dmnd \
-d /export/groups/drangeli/entap_config/bin/uniprot_trembl.dmnd \
--ini /export/groups/drangeli/Jhae.genome/entap_config.ini 
```

The *J. haematoloma* sequences were matched to other Hemiptera, primarily *Halyomorpha halys*, *Cimex lectularius* and *Nilaparavata lugens* for the RefSeq comparisons and *Oncopeltus*, *Lygus* and *Riptortus* from TrEMBL. The contaminant sequence species are also as expected, coming from bacteria and yeasts. Only 12 potential contaminant sequences had homology to humans. 

<a name="sequencing-for-gene-expression-quantification"><a/>

### Sequencing for gene expression quantification

For gene expression analysis, RNA samples were shipped overnight on dry ice to the [DNA Technologies Core](https://dnatech.genomecenter.ucdavis.edu/) at the University of California at Davis. Libraries were prepared using the QuantSeq 3' mRNA-Seq Library Prep Kit (Lexogen, Greenland, New Hampshire, USA) in the three batches for sequencing in separate Illumina HiSeq lanes. Batch 3 was sequenced twice and reads counts for each run were added together for each sample after filtering.

These batches were sequenced in three lanes yielding 115.9×10^9^ bp. These data

Short read sequences were inspected for quality using FastQC (http://www.bioinformatics.babraham.ac.uk/projects/fastqc/), before and after trimmed using [Trimmomatic](http://www.usadellab.org/cms/?page=trimmomatic) version 0.33, as in the example command below.

```bash
java -jar trimmomatic-0.33.jar SE \
raw.data/A1-d25-2_S161_L006_R1_001.fastq \
filtered.reads/A1-d25-2_S161_L006_R1_001.trimmed.fastq \
ILLUMINACLIP:/export/local/src/Trimmomatic-0.33/adapters/TruSeq3-SE.fa:2:30:10 \
HEADCROP:16 LEADING:3 TRAILING:3 SLIDINGWINDOW:3:15 MINLEN:79
```

> A simple [bash script](https://github.com/aphanotus/Jhae.genome/blob/main/trimming.3seq.reads.sh) was used to apply trimming to all read files and save the output in a separate folder.FastQC inspection after trimming revealed notable improvements.
>

**Table S2**: Read numbers before and after quality filtering.

| batch   |     raw reads | filtered reads | passing filter |
| :------ | ------------: | -------------: | --------------:|
| 1       |   341,370,316 |    323,188,378 | 94.7%          |
| 2       |   321,564,969 |    313,590,249 | 97.5%          |
| 3       |   484,588,962 |    419,931,564 | 86.7%          |
| overall | 1,147,524,247 |  1,056,710,191 | 92.1%          |

<a name="mapping"><a/>

### Mapping 3seq reads to the transcriptome

[Kallisto](https://pachterlab.github.io/kallisto/about) was used to quantify the abundance of transcripts from 3seq data using a pseudo-alignment method ([Bray et al. 2016](https://www.nature.com/articles/nbt.3519)). First, a Kallisto index was created from the assembled transcripts, collapsed by cd-hit-est. 

> We chose to use this target, rather than the TransDecoder CDS files, because if there are fragmentary or non-coding RNAs that were sequenced, then some reads will correspond to those RNAs too.

```bash
cd /research/drangeli/phase2_BCG_RNAseq/Jhae_GU_trinity_assembly/
kallisto index --index=GU.Trinity.c90.kallisto.index GU.Trinity.c90.fa
```

In order to run Kallisto on single-end sequence, the program requires the mean and standard deviation of read lengths. The total number of reads was found using bash commands.

```bash
cd /export/groups/drangeli/morphDE/davis.3seq/filtered.reads

gunzip -c 180824/*.fastq.gz 190917/*.fastq.gz 200611/*.fastq.gz 200701/*.fastq.gz \
| sed -n -e '2~4p' \
| awk '{ print length($0); }' > read.lengths.txt

wc -l read.lengths.txt
```

Total read number was 1,149,235,725. 

Mean and  standard deviation were calculated using `awk`.

```bash
cat read.lengths.txt | awk '{for(i=1;i<=NF;i++) {sum[i] += $i; sumsq[i] += ($i)^2}} END {for (i=1;i<=NF;i++) { printf "%f %f \n", sum[i]/NR, sqrt((sumsq[i]-sum[i]^2/NR)/NR)} }'
```

- mean = 84.892739 bp
- SD = 0.507056 bp

A [bash script](https://github.com/aphanotus/Jhae.genome/blob/main/Jhae.kallisto.mapping.sh) was used to run `kallisto quant` for each sample and to rename the output files with the sample names. 

> The result is a folder `kallisto.vs.JhaeGUtx` that contains `.abundance.tsv` files for each sample, listing the counts for each transcript. 

The resulting output files were then merged into a single comma-separated values (CSV) file with columns for each sample.

> This file `kallisto.counts.csv` is 309 Mb, which is too large to post to GitHub.

<a name="differential-gene-expression-analysis"><a/>

### Differential gene expression analysis

Gene expression differences were examined using the Bioconductor package DESeq2 version 1.32.0 ([Love et al. 2014](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0550-8)) in R version 4.1.1 ([R Core Team 2021](https://www.R-project.org/)). The function `DESeq` fits a generalized linear model where counts are modeled using a negative binomial distribution with a gene-specific dispersion parameter. Means were estimated using approximate posterior estimation for generalized linear models ([Zhu et al. 2018](https://academic.oup.com/bioinformatics/article/35/12/2084/5159452)). We extracted log<sub>2</sub> fold-change and p-values corrected by independent hypothesis weighing ([Ignatiadis et al. 2016](https://www.nature.com/articles/nmeth.3885)) and applied a critical threshold of 0.05. Contrasts were made for sex, wing morph and food regime, after accounting for batch effects (multiple sequencing runs). Data were analyzed separately for each stage (fifth instar juveniles and nascent adults) and for each tissue (thorax and gonads). Differential gene expression was visualized by customized plots generated using ggplot2. Variance stabilizing transformation, using the function `vst`,  was applied before ordination using principle component analysis.

> For more details see the R script [`analysis.dge.R`](https://github.com/aphanotus/morphDE/blob/main/analysis.dge.R)

Batch effects were introduced by the need to sequence samples in multiple lanes (Figure S1).

![](https://i.imgur.com/MAjpAP7.jpg)

**Figure S1**. Principle components of uncorrected counts, color-coded by biological factors (A) and sequencing batch (B). 

DESeq2 allows variance from confounding factors, such as sequencing batch, to be accounted for through generalized linear modeling. The normalized counts that result from the model then reflect biologically meaningful variance (Figure S2).

<img src="https://i.imgur.com/XFd0Alf.jpg" style="zoom:25%;" />

**Figure S2**. Principle components of transcript counts for each sample, following normalization for sequencing run (batch). Samples are color-coded by biological origin. 

More complex models can be applied to examine specific contrasts of biological interest. These models partition the variance in gene expression sequentially to factors in the model. For more complex models, less power exists to detect differences, but it allows for the influence of multiple factors to be disentangle from one another. 

#### Expression differences by sex

Sex is often an important factor effecting gene expression. While it is not our focus of study, examining our dataset for differences between females and males in each tissue is a useful confirmation of these expectations (Figure S3. Moreover, sexually dimorphic gene expression has not previously been described in *J. haematoloma*. Genes were differentially expressed at both sampled stages and in both sampled tissues. Adults had a larger number of sexually dimorphic DEGs (84% in gonad; 29% in thorax) compared to juveniles (6.1% in gonads, 0.18% in thorax). The gonads also had more sexually dimorphic gene expression than did the thorax. Within the gonads, males had more significantly biased genes than did females, at both stages (Figure S3C-D). Several genes with roles in somatic sex determination were among these, including *doublesex* (*dsx*) and *transformer* [ref]. The histone acetyltransferase encoded by *mof* is known to increase expression on the X chromosome in *D. melanogaster* males (Smith et al. 2000), although the *mof* ortholog appears up-regulated in the  *J. haematoloma* female thorax (Fig S3B). In the adult gonad contrast, *upd* is significantly biased towards male expression. In *D. melanogaster* testis, *upd* is expressed by hub cells promote maintenance of spermatogenic stem cells (reviewed by Matunis et al 2012).

![](https://i.imgur.com/TejKrlQ.jpg)

**Figure S3**. Volcano plots, showing log<sub>2</sub> fold-change (LFC) and -log<sub>10</sub> FDR-adjusted p-values for contrasts by sex in the juvenile thorax (A), adult thorax (B), juvenile gonads (C) and adult gonads (D). Genes are highlighted in red if they have an absolute LFC values > 1 (two-fold difference) and an FDR-adjusted p-value < 0.05 (panels A) or 10<sup>-3</sup>  (panels B-D). Highlighted transcript are annotated where possible. 

#### Expression differences by morph

> Add here!

![](https://i.imgur.com/UwSOU3H.jpg)

**Figure S4**. Volcano plots, showing log<sub>2</sub> fold-change (LFC) and -log<sub>10</sub> FDR-adjusted p-values from models of gene expression based on correlation to wing shape PC1 in the adult thorax (A), adult gonads (B), adult ovaries (C) and adult testes (D). Greater log-fold change is associated with long-wing shapes. Genes are highlighted in red if they have an absolute LFC values > 1 (two-fold difference) and an FDR-adjusted p-value < 10<sup>-3</sup>  (panels A) or 0.05 (panels B-D). Highlighted transcript are annotated where possible. 

![](https://i.imgur.com/d7rYsNy.jpg)

**Figure S4**. Volcano plots, showing log<sub>2</sub> fold-change (LFC) and -log<sub>10</sub> FDR-adjusted p-values from models of gene expression based on correlation to thorax shape PC1 in the adult thorax (A), adult gonads (B), adult ovaries (C) and adult testes (D). Greater log-fold change is associated with relatively wider thorax shapes. Genes are highlighted in red if they have an absolute LFC values > 1 (two-fold difference) and an FDR-adjusted p-value < 10<sup>-3</sup>  (panels A) or 0.05 (panels B-D). Highlighted transcript are annotated where possible. 

> Not sure if we also want to show the figures for models by morph?

![](https://i.imgur.com/J1UKjhj.jpg)

**Figure SX**. Volcano plots, showing log<sub>2</sub> fold-change (LFC) and -log<sub>10</sub> FDR-adjusted p-values from contrasts by morph in the adult thorax (A), adult gonads (B), adult ovaries (C) and adult testes (D). Greater log-fold change is associated with long-wing morphs. Genes are highlighted in red if they have an absolute LFC values > 1 (two-fold difference) and an FDR-adjusted p-value < 10<sup>-3</sup>  (panels A) or 0.05 (panels B-D). Highlighted transcript are annotated where possible. 

#### Expression differences by food regime

> Add here!

![](https://i.imgur.com/GjGvP7m.jpg)

**Figure SX**. Volcano plots, showing log<sub>2</sub> fold-change (LFC) and -log<sub>10</sub> FDR-adjusted p-values from models of gene expression based on correlation to food density (log~10~ seeds/bug) in the juvenile thorax (A), adult thorax (B), juvenile gonads (C), and adult gonads (D). Greater log-fold change is associated with more available seeds per bug. Genes are highlighted in red if they have an absolute LFC values > 1 (two-fold difference) and an FDR-adjusted p-value < 0.05. Highlighted transcript are annotated where possible. 

#### Expression differences by stage

> Add here!

![](https://i.imgur.com/F2Fmd7N.jpg)

**Figure SX**. Volcano plots, showing log<sub>2</sub> fold-change (LFC) and -log<sub>10</sub> FDR-adjusted p-values for contrasts by stage in the thorax (A), ovaries (B), and testes (C). Genes are highlighted in red if they have an absolute LFC values > 1 (two-fold difference) and an FDR-adjusted p-value < 10<sup>-3</sup>. Highlighted transcript are annotated where possible. 


<a name="enrichment-analysis"><a/>

### Enrichment analysis

Enrichment analyses were performed for differentially expressed genes from each model based on annotations with terms from the GO and KEGG ontologies, as well as EggNOG protein domains, applied to transcripts by EnTAP (Table S3). Genes significant positive and negative expression bias were analyzed separately. We also tested for enrichment of xenic or contaminant transcripts among those that were differentially expressed, and neither of these categories was significantly over-represented. For each term, p-values from the hypergeometric test, implemented in the base R function `phyper`, were adjusted for false discovery rates (FDR). Descriptions of GO terms were obtained using the R package `GO.db` (Carlson, 2021). Names of KEGG pathways can be obtained from the KEGG website via, e.g. `curl http://togows.dbcls.jp/entry/pathway/map04974/name`. This call was automated using the R function `RCurl::getURL` (Temple Lang, 2022).

> For more details on the enrichment analysis see the R script [`enrichment.R`](https://github.com/aphanotus/morphDE/blob/main/enrichment.R)

**Table S3**: Terms from GO and KEGG ontologies and EggNOG protein domains over-represented in differentially expressed genes in each dataset and contrast.

<div style="border: 1px solid #ddd; padding: 5px; overflow-y: scroll; height:500px; table-layout: fixed; overflow-x: scroll; width:925px;">

| stage | tissue | model | bias | ontology | enriched terms | descriptions of top terms (adjusted *p*-values from HGT)                          |
|:----- |:------ |:----- |:---- |:-------- |:--------------:|:---------------------------------------------------------------------- |
| juvenile | thorax | sex | male | GO | 0 |  |
| juvenile | thorax | sex | male | KEGG | 0 |  |
| juvenile | thorax | sex | male | eggNOG | 7 | CDC48_2, 0.0051; CDC48_N, 0.0051; DM, 0.0051; Meckelin, 0.0051; Vps4_C, 0.016; AAA_2, 0.03; RuvB_N, 0.035 |
| juvenile | thorax | sex | female | GO | 0 |  |
| juvenile | thorax | sex | female | KEGG | 0 |  |
| juvenile | thorax | sex | female | eggNOG | 11 | DM, 3e-04; Apis_Csd, 0.0081; CD34_antigen, 0.0081; DFDF, 0.0081; FDF, 0.0081; FFD_TFG, 0.0081; SDP_N, 0.0081; BAT2_N, 0.014; Btz, 0.019; MOZ_SAS, 0.023 |
| adult | thorax | sex | male | GO | 31 | GO:0034061 DNA polymerase activity, 1.1e-14; GO:0003964 RNA-directed DNA polymerase activity, 5.1e-14; GO:0006260 DNA replication, 5.1e-14; GO:0006278 RNA-dependent DNA biosynthetic process, 5.1e-14; GO:0016779 nucleotidyltransferase activity, 2.2e-13; GO:0004523 RNA-DNA hybrid ribonuclease activity, 2e-10; GO:0006259 DNA metabolic process, 4.6e-10; GO:0004521 endoribonuclease activity, 5.3e-10; GO:0016891 endoribonuclease activity, producing 5'-phosphomonoesters, 6.8e-10; GO:0016893 endonuclease activity, active with either ribo- or deoxyribonucleic acids and producing 5'-phosphomonoesters, 7.6e-10 |
| adult | thorax | sex | male | KEGG | 0 |  |
| adult | thorax | sex | male | eggNOG | 7 | RVT_1, 8.7e-14; Exo_endo_phos, 9.5e-11; RnaseH, 1.9e-08; ZnF_C2HC, 4.1e-08; rve, 0.0035; zf-CCHC, 0.0051; DUF1891, 0.0053 |
| adult | thorax | sex | female | GO | 479 | GO:0022626 cytosolic ribosome, 1.2e-45; GO:0006412 translation, 4.9e-43; GO:0010467 gene expression, 9.3e-34; GO:0005622 intracellular anatomical structure, 1.9e-33; GO:0043228 non-membrane-bounded organelle, 2e-30; GO:0043232 intracellular non-membrane-bounded organelle, 2e-30; GO:0044391 ribosomal subunit, 2e-30; GO:0000022 mitotic spindle elongation, 8e-30; GO:0051231 spindle elongation, 2.2e-29; GO:0003735 structural constituent of ribosome, 2.3e-29 |
| adult | thorax | sex | female | KEGG | 3 | map04974 Protein digestion and absorption, 7.2e-07; map04972 Pancreatic secretion, 4.8e-05; map00513 Various types of N-glycan biosynthesis, 0.00071 |
| adult | thorax | sex | female | eggNOG | 30 | CUB, 1.6e-05; COIL, 1.9e-05; LPD_N, 6.9e-05; Vitellogenin_N, 6.9e-05; Trypsin, 9.8e-05; RRM, 0.00015; RRM_1, 0.00015; Tryp_SPc, 0.00016; SIGNAL, 0.00023; DUF1943, 0.00023 |
| juvenile | gonad | sex | male | GO | 583 | GO:0005622 intracellular anatomical structure, 8.2e-26; GO:0043229 intracellular organelle, 2.2e-22; GO:0043231 intracellular membrane-bounded organelle, 1.2e-19; GO:0043227 membrane-bounded organelle, 1.9e-19; GO:0015630 microtubule cytoskeleton, 2.1e-19; GO:0005856 cytoskeleton, 2.8e-16; GO:0016043 cellular component organization, 5.9e-15; GO:0005634 nucleus, 3e-14; GO:0043167 ion binding, 5.4e-14; GO:0048869 cellular developmental process, 5e-13 |
| juvenile | gonad | sex | male | KEGG | 15 | map00710 Carbon fixation in photosynthetic organisms, 0.00091; map01100 Metabolic pathways, 0.0077; map01110 Biosynthesis of secondary metabolites, 0.0077; map04340 Hedgehog signaling pathway, 0.0077; map05110 Vibrio cholerae infection, 0.0077; map05012 Parkinson disease, 0.0097; map00480 Glutathione metabolism, 0.015; map04966 Collecting duct acid secretion, 0.016; map00030 Pentose phosphate pathway, 0.017; map05010 Alzheimer disease, 0.017 |
| juvenile | gonad | sex | male | eggNOG | 32 | COIL, 2.1e-18; S_TKc, 8.2e-06; LRR, 6.8e-05; cNMP_binding, 0.001; Pkinase_Tyr, 0.001; Pkinase, 0.0013; cNMP, 0.0016; RING, 0.0017; S_TK_X, 0.0017; LRR_1, 0.003 |
| juvenile | gonad | sex | female | GO | 1043 | GO:0005737 cytoplasm, 1.2e-59; GO:0005622 intracellular anatomical structure, 5.1e-59; GO:0043229 intracellular organelle, 4.5e-51; GO:0044237 cellular metabolic process, 8.1e-41; GO:0071704 organic substance metabolic process, 1.8e-36; GO:0044238 primary metabolic process, 6.6e-36; GO:0043228 non-membrane-bounded organelle, 8.7e-32; GO:0043232 intracellular non-membrane-bounded organelle, 8.7e-32; GO:0048856 anatomical structure development, 9.8e-32; GO:0016043 cellular component organization, 1.6e-31 |
| juvenile | gonad | sex | female | KEGG | 18 | map01100 Metabolic pathways, 1.1e-10; map05012 Parkinson disease, 2.6e-06; map05010 Alzheimer disease, 2.6e-05; map05152 Tuberculosis, 0.0031; map01110 Biosynthesis of secondary metabolites, 0.0049; map04310 Wnt signaling pathway, 0.0049; map04972 Pancreatic secretion, 0.0049; map04144 Endocytosis, 0.0049; map00051 Fructose and mannose metabolism, 0.018; map05169 Epstein-Barr virus infection, 0.018 |
| juvenile | gonad | sex | female | eggNOG | 16 | TRANS, 3.7e-28; SIGNAL, 1.4e-25; COIL, 9.7e-25; Miro, 7e-04; Arf, 0.0011; Ras, 0.0014; EFh, 0.0016; FN3, 0.0032; GTP_EFTU, 0.0032; fn3, 0.0036 |
| adult | gonad | sex | male | GO | 0 |  |
| adult | gonad | sex | male | KEGG | 0 |  |
| adult | gonad | sex | male | eggNOG | 0 |  |
| adult | gonad | sex | female | GO | 43 | GO:0005622 intracellular anatomical structure, 6.3e-06; GO:0005737 cytoplasm, 6.8e-06; GO:0043229 intracellular organelle, 2.4e-05; GO:0007275 multicellular organism development, 3.1e-05; GO:0016043 cellular component organization, 0.00061; GO:0048731 system development, 0.0011; GO:0048856 anatomical structure development, 0.0023; GO:0043227 membrane-bounded organelle, 0.0091; GO:0043231 intracellular membrane-bounded organelle, 0.0098; GO:0044238 primary metabolic process, 0.0098 |
| adult | gonad | sex | female | KEGG | 0 |  |
| adult | gonad | sex | female | eggNOG | 0 |  |
| juvenile | gonad | sex + wing pad shape | narrow | GO | 0 |  |
| juvenile | gonad | sex + wing pad shape | narrow | KEGG | 0 |  |
| juvenile | gonad | sex + wing pad shape | narrow | eggNOG | 0 |  |
| juvenile | gonad | sex + wing pad shape | wide | GO | 0 |  |
| juvenile | gonad | sex + wing pad shape | wide | KEGG | 3 | map04975 Fat digestion and absorption, 0.0041; map00561 Glycerolipid metabolism, 0.0068; map04972 Pancreatic secretion, 0.015 |
| juvenile | gonad | sex + wing pad shape | wide | eggNOG | 1 | Lipase, 0.009 |
| adult | thorax | sex + wing shape | short wing | GO | 687 | GO:0005622 intracellular anatomical structure, 3.8e-38; GO:0005737 cytoplasm, 1.1e-36; GO:0043229 intracellular organelle, 1e-28; GO:0043227 membrane-bounded organelle, 3.5e-25; GO:0043231 intracellular membrane-bounded organelle, 6.5e-25; GO:0006810 transport, 3e-23; GO:0051234 establishment of localization, 7.3e-23; GO:0071704 organic substance metabolic process, 1.8e-19; GO:0048856 anatomical structure development, 1.1e-18; GO:0050789 regulation of biological process, 1.2e-18 |
| adult | thorax | sex + wing shape | short wing | KEGG | 4 | map04974 Protein digestion and absorption, 1.3e-05; map04972 Pancreatic secretion, 0.0026; map00513 Various types of N-glycan biosynthesis, 0.019; map01100 Metabolic pathways, 0.019 |
| adult | thorax | sex + wing shape | short wing | eggNOG | 26 | TRANS, 1.7e-28; COIL, 5.5e-18; SIGNAL, 5.5e-18; CUB, 4.5e-05; Gtr1_RagA, 0.00045; Miro, 0.0027; Ras, 0.0027; Tubulin_C, 0.0027; Tubulin, 0.007; Arf, 0.014 |
| adult | thorax | sex + wing shape | long wing | GO | 721 | GO:0005739 mitochondrion, 2.3e-110; GO:0005737 cytoplasm, 1.8e-100; GO:0005622 intracellular anatomical structure, 1.3e-79; GO:0043229 intracellular organelle, 4.8e-69; GO:0005759 mitochondrial matrix, 2.8e-58; GO:0043231 intracellular membrane-bounded organelle, 3.2e-57; GO:0043227 membrane-bounded organelle, 3.7e-57; GO:0044237 cellular metabolic process, 5e-54; GO:0000313 organellar ribosome, 6.2e-45; GO:0005761 mitochondrial ribosome, 6.2e-45 |
| adult | thorax | sex + wing shape | long wing | KEGG | 15 | map01100 Metabolic pathways, 2.8e-40; map05012 Parkinson disease, 6.4e-29; map05010 Alzheimer disease, 3.5e-24; map01110 Biosynthesis of secondary metabolites, 2.5e-15; map01120 Microbial metabolism in diverse environments, 8.1e-05; map04260 Cardiac muscle contraction, 0.00043; map00640 Propanoate metabolism, 0.00043; map04710 Circadian rhythm, 0.0037; map00280 Valine, leucine and isoleucine degradation, 0.0093; map01210 2-Oxocarboxylic acid metabolism, 0.0094 |
| adult | thorax | sex + wing shape | long wing | eggNOG | 48 | COIL, 4.1e-24; TRANS, 2.1e-14; I-set, 1.2e-05; IGc2, 1.5e-05; V-set, 0.00017; ig, 0.00026; IG, 0.00026; Pkinase_Tyr, 0.00026; Pkinase, 0.00029; IG_like, 0.00031 |
| adult | gonad | sex + wing shape | short wing | GO | 0 |  |
| adult | gonad | sex + wing shape | short wing | KEGG | 0 |  |
| adult | gonad | sex + wing shape | short wing | eggNOG | 10 | Aa_trans, 0.043; Chitin_bind_4, 0.043; DEAD, 0.043; DEXDc, 0.043; DUF1759, 0.043; Helicase_C, 0.043; HELICc, 0.043; Peptidase_A17, 0.043; RVT_2, 0.043; Sugar_tr, 0.045 |
| adult | gonad | sex + wing shape | long wing | GO | 114 | GO:0004197 cysteine-type endopeptidase activity, 0.00011; GO:0005764 lysosome, 0.00011; GO:0008219 cell death, 0.00011; GO:0035070 salivary gland histolysis, 0.00011; GO:0035071 salivary gland cell autophagic cell death, 0.00011; GO:0045169 fusome, 0.00011; GO:0048102 autophagic cell death, 0.00011; GO:0000323 lytic vacuole, 0.00012; GO:0008234 cysteine-type peptidase activity, 0.00017; GO:0007435 salivary gland morphogenesis, 0.00019 |
| adult | gonad | sex + wing shape | long wing | KEGG | 5 | map04612 Antigen processing and presentation, 1.1e-06; map04145 Phagosome, 1.7e-06; map05205 Proteoglycans in cancer, 4.4e-06; map05323 Rheumatoid arthritis, 0.0041; map04142 Lysosome, 0.012 |
| adult | gonad | sex + wing shape | long wing | eggNOG | 9 | Inhibitor_I29, 4.3e-07; Pept_C1, 4.3e-07; Peptidase_C1, 4.3e-07; Peptidase_C1_2, 4.3e-07; SIGNAL, 0.00084; Thiol_cytolysin, 0.0035; Lipase_2, 0.0038; Lipase, 0.0073; TRANS, 0.008 |
| adult | ovaries | sex + wing shape | short wing | GO | 0 |  |
| adult | ovaries | sex + wing shape | short wing | KEGG | 0 |  |
| adult | ovaries | sex + wing shape | short wing | eggNOG | 6 | MIG-14_Wnt-bd, 0.0079; Pacifastin_I, 0.016; Chitin_bind_4, 0.033; DUF1759, 0.033; Peptidase_A17, 0.033; RVT_2, 0.033 |
| adult | ovaries | sex + wing shape | long wing | GO | 41 | GO:0009268 response to pH, 0.011; GO:0010269 response to selenium ion, 0.011; GO:0010447 response to acidic pH, 0.011; GO:0030682 mitigation of host defenses by symbiont, 0.011; GO:0052173 response to defenses of other organism, 0.011; GO:0052200 response to host defenses, 0.011; GO:0052572 response to host immune response, 0.011; GO:0071291 cellular response to selenium ion, 0.011; GO:0075136 response to host, 0.011; GO:0008121 ubiquinol-cytochrome-c reductase activity, 0.015 |
| adult | ovaries | sex + wing shape | long wing | KEGG | 0 |  |
| adult | ovaries | sex + wing shape | long wing | eggNOG | 2 | Sod_Fe_C, 0.0024; Sod_Fe_N, 0.0024 |
| adult | testes | sex + wing shape | short wing | GO | 0 |  |
| adult | testes | sex + wing shape | short wing | KEGG | 0 |  |
| adult | testes | sex + wing shape | short wing | eggNOG | 4 | DEAD, 0.021; DEXDc, 0.021; Helicase_C, 0.021; HELICc, 0.021 |
| adult | testes | sex + wing shape | long wing | GO | 157 | GO:0000131 incipient cellular bud site, 0.029; GO:0000148 1,3-beta-D-glucan synthase complex, 0.029; GO:0000935 division septum, 0.029; GO:0001837 epithelial to mesenchymal transition, 0.029; GO:0004028 3-chloroallyl aldehyde dehydrogenase activity, 0.029; GO:0004030 aldehyde dehydrogenase [NAD(P)+] activity, 0.029; GO:0005773 vacuole, 0.029; GO:0005934 cellular bud tip, 0.029; GO:0005937 mating projection, 0.029; GO:0006068 ethanol catabolic process, 0.029 |
| adult | testes | sex + wing shape | long wing | KEGG | 0 |  |
| adult | testes | sex + wing shape | long wing | eggNOG | 18 | Lys, 0.015; LYZ1, 0.015; Peptidase_C13, 0.015; SIGNAL, 0.015; SLT, 0.015; Thyroglobulin_1, 0.015; Transglycosylas, 0.015; TY, 0.015; LuxC, 0.02; CY, 0.022 |
| adult | thorax | sex + thorax shape | wide | GO | 0 |  |
| adult | thorax | sex + thorax shape | wide | KEGG | 0 |  |
| adult | thorax | sex + thorax shape | wide | eggNOG | 11 | SIGNAL, 0.00011; CH, 0.0036; CAMSAP_CH, 0.01; CRAL_TRIO, 0.013; JHBP, 0.013; Calponin, 0.013; Polysacc_deac_1, 0.013; SEC14, 0.013; TRANS, 0.013; CRAL_TRIO_N, 0.026 |
| adult | thorax | sex + thorax shape | narrow | GO | 0 |  |
| adult | thorax | sex + thorax shape | narrow | KEGG | 0 |  |
| adult | thorax | sex + thorax shape | narrow | eggNOG | 16 | DNA_pol_B_2, 0.047; DUF1162, 0.047; DUF1986, 0.047; FYRC, 0.047; FYRN, 0.047; IR1-M, 0.047; KASH, 0.047; Neur_chan_LBD, 0.047; Neur_chan_memb, 0.047; Ran_BP1, 0.047 |
| juvenile | thorax | sex + morph + food density | more food | GO | 0 |  |
| juvenile | thorax | sex + morph + food density | more food | KEGG | 0 |  |
| juvenile | thorax | sex + morph + food density | more food | eggNOG | 6 | Lipocalin, 0.0057; Lipocalin_2, 0.0057; Rad50_zn_hook, 0.0057; Triabin, 0.0057; VWC_def, 0.012; VWC, 0.013 |
| juvenile | thorax | sex + morph + food density | less food | GO | 8 | GO:0003916 DNA topoisomerase activity, 0.046; GO:0003917 DNA topoisomerase type I (single strand cut, ATP-independent) activity, 0.046; GO:0003918 DNA topoisomerase type II (double strand cut, ATP-hydrolyzing) activity, 0.046; GO:0005160 transforming growth factor beta receptor binding, 0.046; GO:0006265 DNA topological change, 0.046; GO:0016191 synaptic vesicle uncoating, 0.046; GO:0072319 vesicle uncoating, 0.046 |
| juvenile | thorax | sex + morph + food density | less food | KEGG | 0 |  |
| juvenile | thorax | sex + morph + food density | less food | eggNOG | 8 | DUF3480, 0.0013; SARA, 0.0013; TOPEUc, 0.0013; Topoisom_I, 0.0013; Topoisom_I_N, 0.0013; FYVE, 0.013; MADF_DNA_bdg, 0.014; DnaJ, 0.02 |
| adult | thorax | sex + morph + food density | more food | GO | 10 | GO:0006801 superoxide metabolic process, 0.0018; GO:0072593 reactive oxygen species metabolic process, 0.0032; GO:0005319 lipid transporter activity, 0.0043; GO:0006869 lipid transport, 0.0071; GO:0010876 lipid localization, 0.01; GO:0047396 glycosylphosphatidylinositol diacylglycerol-lyase activity, 0.015; GO:0033036 macromolecule localization, 0.024; GO:0045445 myoblast differentiation, 0.024; GO:0000083 regulation of transcription involved in G1/S transition of mitotic cell cycle, 0.031 |
| adult | thorax | sex + morph + food density | more food | KEGG | 0 |  |
| adult | thorax | sex + morph + food density | more food | eggNOG | 10 | DUF1943, 4.1e-05; LPD_N, 4.1e-05; Sod_Cu, 4.1e-05; Vitellogenin_N, 4.1e-05; VWD, 4.6e-05; PI-PLC-X, 0.0066; SIGNAL, 0.015; Tryp_SPc, 0.035; Trypsin, 0.035; zf-C2H2, 0.036 |
| adult | thorax | sex + morph + food density | less food | GO | 0 |  |
| adult | thorax | sex + morph + food density | less food | KEGG | 2 | map04391 Hippo signaling pathway - fly, 0.044; map04660 T cell receptor signaling pathway, 0.044 |
| adult | thorax | sex + morph + food density | less food | eggNOG | 8 | 6PF2K, 0.0095; KTI12, 0.0095; L27_1, 0.0095; Guanylate_kin, 0.012; GuKc, 0.012; L27, 0.012; Sulfotransfer_1, 0.012; DUF1758, 0.02 |
| juvenile | gonad | sex + morph + food density | more food | GO | 57 | GO:0004044 amidophosphoribosyltransferase activity, 0.01; GO:0004360 glutamine-fructose-6-phosphate transaminase (isomerizing) activity, 0.01; GO:0016627 oxidoreductase activity, acting on the CH-CH group of donors, 0.01; GO:0016628 oxidoreductase activity, acting on the CH-CH group of donors, NAD or NADP as acceptor, 0.01; GO:0030870 Mre11 complex, 0.01; GO:0032440 2-alkenal reductase [NAD(P)+] activity, 0.01; GO:0070548 L-glutamine aminotransferase activity, 0.01; GO:0005622 intracellular anatomical structure, 0.012; GO:0035079 polytene chromosome puffing, 0.012; GO:0035080 heat shock-mediated polytene chromosome puffing, 0.012 |
| juvenile | gonad | sex + morph + food density | more food | KEGG | 10 | map04010 MAPK signaling pathway, 0.014; map04141 Protein processing in endoplasmic reticulum, 0.014; map04144 Endocytosis, 0.014; map04612 Antigen processing and presentation, 0.014; map05134 Legionellosis, 0.014; map05145 Toxoplasmosis, 0.014; map05162 Measles, 0.014; map05164 Influenza A, 0.014; map00520 Amino sugar and nucleotide sugar metabolism, 0.014; map04915 Estrogen signaling pathway, 0.015 |
| juvenile | gonad | sex + morph + food density | more food | eggNOG | 10 | DWB, 0.0021; GATase_2, 0.0021; IRF-3, 0.0021; MH2, 0.0021; Rad50_zn_hook, 0.0021; SIS, 0.0021; Collagen, 0.0028; HSP70, 0.0028; MreB_Mbl, 0.0028; COIL, 0.016 |
| juvenile | gonad | sex + morph + food density | less food | GO | 34 | GO:0007278 pole cell fate determination, 0.0093; GO:0019090 mitochondrial rRNA export from mitochondrion, 0.0093; GO:0019093 mitochondrial RNA localization, 0.0093; GO:0030719 P granule organization, 0.014; GO:0007277 pole cell development, 0.031; GO:0043186 P granule, 0.031; GO:0045495 pole plasm, 0.031; GO:0060293 germ plasm, 0.031; GO:0008138 protein tyrosine/serine/threonine phosphatase activity, 0.036; GO:0035770 ribonucleoprotein granule, 0.036 |
| juvenile | gonad | sex + morph + food density | less food | KEGG | 0 |  |
| juvenile | gonad | sex + morph + food density | less food | eggNOG | 5 | SMN, 0.004; DSPc, 0.0043; PTPc_DSPc, 0.0043; TUDOR, 0.0043; Y_phosphatase, 0.0048 |
| adult | gonad | sex + morph + food density | more food | GO | 41 | GO:0006590 thyroid hormone generation, 0.0086; GO:0006768 biotin metabolic process, 0.0086; GO:0008523 sodium-dependent multivitamin transmembrane transporter activity, 0.0086; GO:0015111 iodide transmembrane transporter activity, 0.0086; GO:0015705 iodide transport, 0.0086; GO:0015878 biotin transport, 0.0086; GO:0015887 pantothenate transmembrane transport, 0.0086; GO:0015939 pantothenate metabolic process, 0.0086; GO:0035461 vitamin transmembrane transport, 0.0086; GO:0042403 thyroid hormone metabolic process, 0.0086 |
| adult | gonad | sex + morph + food density | more food | KEGG | 0 |  |
| adult | gonad | sex + morph + food density | more food | eggNOG | 6 | IF2_N, 0.0037; SSF, 0.0037; zf-C2HC, 0.0037; MOZ_SAS, 0.0042; LIM, 0.012; AT_hook, 0.024 |
| adult | gonad | sex + morph + food density | less food | GO | 4 | GO:0004446 inositol-hexakisphosphate phosphatase activity, 0.0073; GO:0052826 inositol hexakisphosphate 2-phosphatase activity, 0.0073; GO:0003993 acid phosphatase activity, 0.018; GO:0052745 inositol phosphate phosphatase activity, 0.018 |
| adult | gonad | sex + morph + food density | less food | KEGG | 2 | map04972 Pancreatic secretion, 0.0088; map04974 Protein digestion and absorption, 0.0088 |
| adult | gonad | sex + morph + food density | less food | eggNOG | 10 | Acid_phosphat_A, 0.0029; CUB, 0.011; CwfJ_C_1, 0.011; CwfJ_C_2, 0.011; Mak10, 0.011; SIGNAL, 0.011; Tryp_SPc, 0.011; Trypsin, 0.011; Allatostatin, 0.018; zf-MYND, 0.045 |
| juvenile | ovaries | sex + morph + food density | more food | GO | 144 | GO:0000491 small nucleolar ribonucleoprotein complex assembly, 0.015; GO:0000492 box C/D snoRNP assembly, 0.015; GO:0001884 pyrimidine nucleoside binding, 0.015; GO:0002134 UTP binding, 0.015; GO:0002135 CTP binding, 0.015; GO:0007008 outer mitochondrial membrane organization, 0.015; GO:0009532 plastid stroma, 0.015; GO:0009570 chloroplast stroma, 0.015; GO:0009908 flower development, 0.015; GO:0010658 striated muscle cell apoptotic process, 0.015 |
| juvenile | ovaries | sex + morph + food density | more food | KEGG | 5 | map04621 NOD-like receptor signaling pathway, 0.031; map04626 Plant-pathogen interaction, 0.031; map04612 Antigen processing and presentation, 0.039; map04914 Progesterone-mediated oocyte maturation, 0.039; map04915 Estrogen signaling pathway, 0.039 |
| juvenile | ovaries | sex + morph + food density | more food | eggNOG | 2 | HATPase_c, 0.0099; HSP90, 0.0099 |
| juvenile | ovaries | sex + morph + food density | less food | GO | 0 |  |
| juvenile | ovaries | sex + morph + food density | less food | KEGG | 1 | map00520 Amino sugar and nucleotide sugar metabolism, 0.039 |
| juvenile | ovaries | sex + morph + food density | less food | eggNOG | 25 | A2M, 0.0035; A2M_comp, 0.0035; A2M_N, 0.0035; A2M_N_2, 0.0035; A2M_recep, 0.0035; DMPK_coil, 0.0035; Thiol-ester_cl, 0.0035; GATase_2, 0.0045; HyaE, 0.0045; PBD, 0.0045 |
| adult | ovaries | sex + morph + food density | more food | GO | 54 | GO:0005319 lipid transporter activity, 1.4e-10; GO:0006869 lipid transport, 3.3e-10; GO:0006801 superoxide metabolic process, 4.6e-10; GO:0010876 lipid localization, 2e-09; GO:0072593 reactive oxygen species metabolic process, 2e-09; GO:0071702 organic substance transport, 3.5e-05; GO:0033036 macromolecule localization, 0.00098; GO:0006810 transport, 0.0055; GO:0051234 establishment of localization, 0.006; GO:0006590 thyroid hormone generation, 0.022 |
| adult | ovaries | sex + morph + food density | more food | KEGG | 0 |  |
| adult | ovaries | sex + morph + food density | more food | eggNOG | 16 | DUF1943, 3.3e-12; LPD_N, 3.3e-12; Vitellogenin_N, 3.3e-12; Sod_Cu, 5.6e-12; VWD, 8.9e-12; DUF3608, 0.0043; MENTAL, 0.0043; SIGNAL, 0.0043; Skp1, 0.0069; Skp1_POZ, 0.0069 |
| adult | ovaries | sex + morph + food density | less food | GO | 0 |  |
| adult | ovaries | sex + morph + food density | less food | KEGG | 0 |  |
| adult | ovaries | sex + morph + food density | less food | eggNOG | 7 | Acid_phosphat_A, 0.043; Chromo, 0.043; CHROMO, 0.043; RhoGAP, 0.043; small_GTPase, 0.043; zf-MYND, 0.043; zf-UBR, 0.043 |
| juvenile | testes | sex + morph + food density | more food | GO | 0 |  |
| juvenile | testes | sex + morph + food density | more food | KEGG | 0 |  |
| juvenile | testes | sex + morph + food density | more food | eggNOG | 4 | COX15-CtaA, 0.026; Mlf1IP, 0.026; N-SET, 0.026; Ssl1, 0.026 |
| juvenile | testes | sex + morph + food density | less food | GO | 0 |  |
| juvenile | testes | sex + morph + food density | less food | KEGG | 0 |  |
| juvenile | testes | sex + morph + food density | less food | eggNOG | 19 | Acyl_transf_1, 0.028; Acyl_transf_3, 0.028; HELICc3, 0.028; HSA, 0.028; Jnk-SapK_ap_N, 0.028; Ketoacyl-synt_C, 0.028; NRF, 0.028; PKS_ER, 0.028; PKS_KR, 0.028; PKS_KS, 0.028 |
| adult | testes | sex + morph + food density | more food | GO | 6 | GO:0007278 pole cell fate determination, 0.019; GO:0019090 mitochondrial rRNA export from mitochondrion, 0.019; GO:0019093 mitochondrial RNA localization, 0.019; GO:0030719 P granule organization, 0.019; GO:0042479 positive regulation of eye photoreceptor cell development, 0.019; GO:0045315 positive regulation of compound eye photoreceptor development, 0.019 |
| adult | testes | sex + morph + food density | more food | KEGG | 4 | map04360 Axon guidance, 0.026; map04510 Focal adhesion, 0.026; map04660 T cell receptor signaling pathway, 0.026; map04810 Regulation of actin cytoskeleton, 0.026 |
| adult | testes | sex + morph + food density | more food | eggNOG | 4 | PBD, 0.0082; SMN, 0.0082; RhoGAP, 0.02; TUDOR, 0.02 |
| adult | testes | sex + morph + food density | less food | GO | 1 |  |
| adult | testes | sex + morph + food density | less food | KEGG | 0 |  |
| adult | testes | sex + morph + food density | less food | eggNOG | 5 | dDENN, 0.0015; DENN, 0.0015; DUF1630, 0.0015; Mlf1IP, 0.0015; uDENN, 0.0015 |
|  | thorax | sex + stage | juvenile | GO | 1437 | GO:0005622 intracellular anatomical structure, 7e-96; GO:0071704 organic substance metabolic process, 5.8e-94; GO:0044238 primary metabolic process, 7.2e-88; GO:0043229 intracellular organelle, 4.9e-87; GO:0044237 cellular metabolic process, 1.3e-75; GO:0043227 membrane-bounded organelle, 1.4e-64; GO:0043231 intracellular membrane-bounded organelle, 1.5e-64; GO:0005737 cytoplasm, 1.3e-63; GO:0043170 macromolecule metabolic process, 1.4e-52; GO:0006807 nitrogen compound metabolic process, 2.7e-47 |
|  | thorax | sex + stage | juvenile | KEGG | 74 | map01100 Metabolic pathways, 2e-28; map01110 Biosynthesis of secondary metabolites, 8.4e-10; map04145 Phagosome, 5e-08; map00500 Starch and sucrose metabolism, 1.1e-06; map00982 Drug metabolism - cytochrome P450, 8.5e-06; map00980 Metabolism of xenobiotics by cytochrome P450, 9.9e-06; map03410 Base excision repair, 1.5e-05; map03420 Nucleotide excision repair, 1.5e-05; map00053 Ascorbate and aldarate metabolism, 1.8e-05; map04972 Pancreatic secretion, 2e-05 |
|  | thorax | sex + stage | juvenile | eggNOG | 64 | TRANS, 4.1e-75; SIGNAL, 5.7e-56; COIL, 6.6e-40; MFS_1, 8.2e-12; RRM_1, 2.7e-10; RRM, 8.3e-10; Sugar_tr, 1.3e-09; A1_Propeptide, 7.7e-07; Chitin_bind_4, 9.5e-07; Asp, 1.1e-06 |
|  | thorax | sex + stage | adult | GO | 618 | GO:0005739 mitochondrion, 2.9e-24; GO:0005740 mitochondrial envelope, 7.9e-19; GO:0031966 mitochondrial membrane, 1.7e-18; GO:0005622 intracellular anatomical structure, 2.3e-18; GO:0005737 cytoplasm, 1.5e-17; GO:0048856 anatomical structure development, 1.8e-17; GO:0042302 structural constituent of cuticle, 7.9e-16; GO:0005743 mitochondrial inner membrane, 8.8e-16; GO:0019866 organelle inner membrane, 1.7e-15; GO:0031967 organelle envelope, 2.1e-15 |
|  | thorax | sex + stage | adult | KEGG | 4 | map01100 Metabolic pathways, 1.2e-15; map05012 Parkinson disease, 1.4e-14; map05010 Alzheimer disease, 1.9e-10; map01110 Biosynthesis of secondary metabolites, 0.0071 |
|  | thorax | sex + stage | adult | eggNOG | 33 | TRANS, 9.7e-45; SIGNAL, 6.2e-42; Chitin_bind_4, 8.8e-15; COIL, 8.9e-15; Sod_Cu, 0.0012; LIM, 0.0025; IG, 0.0035; CBM_14, 0.0072; ChtBD2, 0.0072; LDLa, 0.0091 |
|  | ovaries | sex + stage | juvenile | GO | 5 | GO:0034061 DNA polymerase activity, 4.8e-06; GO:0003964 RNA-directed DNA polymerase activity, 1.5e-05; GO:0006278 RNA-dependent DNA biosynthetic process, 1.5e-05; GO:0006260 DNA replication, 0.00014; GO:0016779 nucleotidyltransferase activity, 0.0085 |
|  | ovaries | sex + stage | juvenile | KEGG | 0 |  |
|  | ovaries | sex + stage | juvenile | eggNOG | 5 | RVT_1, 5e-07; ZnF_C2HC, 0.016; Exo_endo_phos, 0.017; rve, 0.033; DUF1759, 0.038 |
|  | ovaries | sex + stage | adult | GO | 190 | GO:0005634 nucleus, 2.7e-19; GO:0043229 intracellular organelle, 4.3e-12; GO:0090304 nucleic acid metabolic process, 7.4e-12; GO:0043227 membrane-bounded organelle, 8.6e-12; GO:0043231 intracellular membrane-bounded organelle, 8.6e-12; GO:0005622 intracellular anatomical structure, 1.2e-11; GO:0031981 nuclear lumen, 1.3e-11; GO:1901360 organic cyclic compound metabolic process, 3.4e-11; GO:0006139 nucleobase-containing compound metabolic process, 5.3e-11; GO:0006725 cellular aromatic compound metabolic process, 5.3e-11 |
|  | ovaries | sex + stage | adult | KEGG | 0 |  |
|  | ovaries | sex + stage | adult | eggNOG | 1 | COIL, 2.6e-05 |
|  | testes | sex + stage | juvenile | GO | 3 | GO:0035070 salivary gland histolysis, 0.034; GO:0035071 salivary gland cell autophagic cell death, 0.034; GO:0048102 autophagic cell death, 0.034 |
|  | testes | sex + stage | juvenile | KEGG | 0 |  |
|  | testes | sex + stage | juvenile | eggNOG | 0 |  |
|  | testes | sex + stage | adult | GO | 0 |  |
|  | testes | sex + stage | adult | KEGG | 0 |  |
|  | testes | sex + stage | adult | eggNOG | 0 |  |

</div>

---

Return to https://github.com/aphanotus/morphDE
